prow/bugzilla/client_test.go

```diff
@@ -1004,74 +1004,78 @@ func errorChecker(err error, t *testing.T) {
 	}
 }
 func TestGetAllClones(t *testing.T) {
-	fake := &Fake{}
-	fake.Bugs = map[int]Bug{}
-	fake.BugComments = map[int][]Comment{}
-	bug1Create := &BugCreate{
-		Summary: "Dummy bug to test getAllClones",
-	}
-	bug1ID, err := fake.CreateBug(bug1Create)
-	if err != nil {
-		t.Fatalf("Error while creating bug in Fake!\n")
-	}
-	bug1, err := fake.GetBug(bug1ID) //Original bug
-	errorChecker(err, t)
-	bug2ID, err := fake.CloneBug(bug1) //2nd level bug
-	errorChecker(err, t)
-	bug2, err := fake.GetBug(bug2ID)
-	errorChecker(err, t)
-	bug3ID, err := fake.CloneBug(bug2) //3rd level bug
-	errorChecker(err, t)
-	bug4ID, err := fake.CloneBug(bug1) //Sibling of 2nd level
-	errorChecker(err, t)
-	bug1, err = fake.GetBug(bug1ID)
-	errorChecker(err, t)
-	bug2, err = fake.GetBug(bug2ID)
-	errorChecker(err, t)
-	bug3, err := fake.GetBug(bug3ID)
-	errorChecker(err, t)
-	bug4, err := fake.GetBug(bug4ID)
-	errorChecker(err, t)
+
 	testcases := []struct {
 		name           string
-		bug            *Bug
-		expectedClones sets.Int
+		bugs           []Bug
+		expectedClones map[int]sets.Int
 	}{
 		{
 			"Clones including multiple children",
-			bug1,
-			sets.NewInt(bug2ID, bug3ID, bug4ID),
-		},
-		{
-			"Clones should include parent as well as child",
-			bug2,
-			sets.NewInt(bug1ID, bug3ID, bug4ID),
-		},
-		{
-			"Clones includes parent and grandparent",
-			bug3,
-			sets.NewInt(bug1ID, bug2ID, bug4ID),
-		},
-		{
-			"Clones when not directly related",
-			bug4,
-			sets.NewInt(bug1ID, bug2ID, bug3ID),
+			[]Bug{
+				{
+					Summary:   "Dummy bug to test getAllClones",
+					ID:        1,
+					DependsOn: []int{},
+					Blocks:    []int{2, 5},
+				},
+				{
+					Summary:   "Dummy bug to test getAllClones",
+					ID:        2,
+					DependsOn: []int{1},
+					Blocks:    []int{3},
+				},
+				{
+					Summary:   "Dummy bug to test getAllClones",
+					ID:        3,
+					DependsOn: []int{2},
+					Blocks:    []int{},
+				},
+				{
+					Summary:   "Not a clone",
+					ID:        4,
+					DependsOn: []int{1},
+				},
+				{
+					Summary:   "Dummy bug to test getAllClones",
+					ID:        5,
+					DependsOn: []int{1},
+				},
+			},
+			map[int]sets.Int{
```

Comment:
Why is this keyed by bug? It should be sufficient to compare a set of bugIDs (without any keying) to the result of `getClones` or not?

Reply:
For the case of bug 4 which doesn't have any clones in this configuration, we would need a separate set of clones - which is why it is keyed

Reply:
I don't follow. It just shouldn't be returned by `getCloenes`, because its not a clone?

Reply:
Yes, bug4 is not a clone

Reply:
Yeah so whats the point of having a set of each bug that is a clone that contains all bugs that are a clone and a set of nothing for the bug that is not a clone? Just use a set of all bugs you expect `getAllClones` to return intstead.

Reply:
I would then have to add additional logic inside the test to identify the one bug which is not a clone and which ones are clones. With this approach I can just configure the clones in the map and not have to do anything else

Reply:
No. Here on the testcase you define a set of expected clones (`sets.NewInt(1,2,3,5)` and in the test logic you construct a new set and compare:
```
				clones, err := getAllClones(fake, &bug, bugCache)
￼				if err != nil {
￼					t.Errorf("Error occurred when none was expected: %v", err)
￼				}
        actualClonesSet := sets.Int{}
        for _, clone := range clones { actualClonesSet.Insert(clone.ID }
        if !tc.expectedClones.Equals(actualCloneSet) { t.Errorf("returned clones do not match expected ones")}

Reply:
Right now I am testing for all the bugs which I am passing in []bugs, do you want me to change that to test for only one bug per testcase? In this case I wouldn't need key the expected clones.

Reply:
There is only one testcase already. I don't undrstand the question. The testcase tests the in and out of the func. Right now you use a super complicated way to test the out of the function. There is no need for that.